### PR TITLE
Tools: add copter gps_glitch_loiter_test2

### PR DESCRIPTION
**Pull request by rmackay9**
This new autotest checks if the vehicle twitches after a longer term (e.g. > 7 second) GPS glitch is applied.  In these cases the position controller should consume the EKF position reset and adjust the targets so that the vehicle's attitude remains stable.

The test is:

- takeoff in Loiter and wait for the roll and pitch attitude to reach nearly zero
- apply a GPS glitch of about 100m
- check the roll and pitch angles remain within 2deg for 20 seconds
- revert  the GPS glitch and RTL

This is quite different from the existing GPS Loiter autotest that ensures the vehicle does not move by more than 20m.  The existing test was not enough to catch a bug that was recently fixed by PR https://github.com/ArduPilot/ardupilot/pull/17850

This has been tested to ensure that it fails when the above PR is reverted and succeeds against latest.

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -1561,6 +1561,34 @@ def fly_gps_glitch_loiter_test(self, timeout=30, max_distance=20):
...
+            m = self.mav.recv_match(type='ATTITUDE', blocking=True)
+            roll_deg = math.degrees(m.roll)
+            pitch_deg = math.degrees(m.pitch)
+            self.progress("checking att: roll=%f pitch=%f " % (roll_deg, pitch_deg))
+            if abs(roll_deg) > 2 or abs(pitch_deg) > 2:
+                raise NotAchievedException("fly_gps_glitch_loiter_test2 failed, roll or pitch moved during GPS glitch")
+                continue
+
+        # restore glitch
+        self.set_parameter("SIM_GPS_GLITCH_X", 0.001)
```
Comment (tridge) [inline Tools/autotest/arducopter.py:1586]: should be zero

**Review by tridge — CHANGES_REQUESTED** [review]

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -1561,6 +1561,34 @@ def fly_gps_glitch_loiter_test(self, timeout=30, max_distance=20):
...
+        # check lean angles remain stable for 20 seconds
+        tstart = self.get_sim_time()
+        while self.get_sim_time_cached() - tstart < 20:
+            m = self.mav.recv_match(type='ATTITUDE', blocking=True)
+            roll_deg = math.degrees(m.roll)
+            pitch_deg = math.degrees(m.pitch)
+            self.progress("checking att: roll=%f pitch=%f " % (roll_deg, pitch_deg))
+            if abs(roll_deg) > 2 or abs(pitch_deg) > 2:
+                raise NotAchievedException("fly_gps_glitch_loiter_test2 failed, roll or pitch moved during GPS glitch")
+                continue
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:1583]: Won't be reached

File: Tools/autotest/arducopter.py (hunk 1 above)
Comment (peterbarker) [inline Tools/autotest/arducopter.py:1586]: context push/pop insteaad

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: Tools/autotest/arducopter.py (hunk 2 above)
Comment (rmackay9) [reply Tools/autotest/arducopter.py]: txs, removed.

File: Tools/autotest/arducopter.py [hunk 3]
```diff
@@ -1561,6 +1561,34 @@ def fly_gps_glitch_loiter_test(self, timeout=30, max_distance=20):
...
+        tstart = self.get_sim_time()
+        while self.get_sim_time_cached() - tstart < 20:
+            m = self.mav.recv_match(type='ATTITUDE', blocking=True)
+            roll_deg = math.degrees(m.roll)
+            pitch_deg = math.degrees(m.pitch)
+            self.progress("checking att: roll=%f pitch=%f " % (roll_deg, pitch_deg))
+            if abs(roll_deg) > 2 or abs(pitch_deg) > 2:
+                raise NotAchievedException("fly_gps_glitch_loiter_test2 failed, roll or pitch moved during GPS glitch")
+
+        # restore glitch
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:1585]: Comment out of place

**Review by peterbarker — APPROVED** [review]

File: Tools/autotest/arducopter.py (hunk 1 above)
Comment (rmackay9) [reply Tools/autotest/arducopter.py]: fixed

Comment (rmackay9) [reply Tools/autotest/arducopter.py]: used push/pop

File: Tools/autotest/arducopter.py (hunk 3 above)
Comment (rmackay9) [reply Tools/autotest/arducopter.py]: fixed up comment, txs

**Pull request merged**